### ISSUE TITLE
[FR] Remove Rule Type Change Restriction

### ISSUE THREAD

**Issue by terrancedejesus** [elastic/detection-rules#2765]
## Summary
We should revisit removing rule type change restrictions from rule tuning. At this time, rule type changes will fail unit tests to avoid a previous bug in older stack versions. This would allow rule authors more flexibility with rule tuning to adjust rules as need be based on features that may only exist in specific rule types.

Example: https://github.com/elastic/detection-rules/pull/2760

**Comment by terrancedejesus** [comment]
@spong - Any insight from the detection engine and Kibana side you can provide on this when you have time. Thanks!

**Comment by spong** [comment]
So we once again started supporting rule type changes back in `8.2` with https://github.com/elastic/kibana/pull/128283, however this was just for the pre-packaged rules route it looks like and I'm not sure if this logic has been carried over to the new routes we're working. @xcrzx, will this behavior still be supported in the new routes?

**Comment by xcrzx** [comment]
> So we once again started supporting rule type changes back in `8.2` with [elastic/kibana#128283](https://github.com/elastic/kibana/pull/128283), however this was just for the pre-packaged rules route it looks like and I'm not sure if this logic has been carried over to the new routes we're working. @xcrzx, will this behavior still be supported in the new routes?

Yup, I'm reusing the rule upgrade logic in the new routes, so the rule type change will be supported.

**Comment by xcrzx** [comment]
> @xcrzx Our rules at a minimum only backport to 8.3, so if we did change rule types for rules it would backport to 8.3 min-stacked rules. However, we only release current-3 so as far as 8.8, we are only releasing packages for 8.5+. Does this by chance help with this at all?

The API I mentioned is not public (it is hidden under a feature flag) and is currently under development, so we are free to introduce any changes to it. But our current public API has supported rule type changes since `8.2`. So if you want to release a package that includes rules with changed types, that is fine and should not affect our current work.

**Comment by Mikaayenson** [comment]
Report of broken changes upstream: https://github.com/elastic/kibana/issues/161094
